# SEC Registrant Analysis

Analyze unique registrant names from SEC metadata

In [15]:
import pandas as pd
import numpy as np
from pathlib import Path

# Configure pandas display
pd.set_option('display.max_rows', 100)
pd.set_option('display.max_columns', None)
pd.set_option('display.width', None)

## Load Data

In [16]:
# Path to CSV file
csv_path = Path('../output/registrant_metadata_2025.csv')

# Load the CSV
df = pd.read_csv(csv_path)

print(f"Total rows: {len(df):,}")
print(f"\nColumns: {list(df.columns)}")
df.head()

Total rows: 1,231,194

Columns: ['registrant_name', 'cik', 'accession_number', 'accession_number_nodashes', 'form_type', 'filing_date', 'file_name', 'source_quarter', 'file_path']


,registrant_name,cik,accession_number,accession_number_nodashes,form_type,filing_date,file_name,source_quarter,file_path
0,&PARTNERS,107136,0001214659-25-002647,121465925002647,13F-HR,2025-02-14,0001214659-25-002647.txt,2025-Q1,edgar/data/107136/0001214659-25-002647.txt
1,&PARTNERS,107136,0000107136-25-000003,10713625000003,MA-A,2025-03-05,0000107136-25-000003.txt,2025-Q1,edgar/data/107136/0000107136-25-000003.txt
2,&PARTNERS,107136,0000107136-25-000005,10713625000005,X-17A-5,2025-03-18,0000107136-25-000005.txt,2025-Q1,edgar/data/107136/0000107136-25-000005.txt
3,"025 - Series of IPOSharks Venture Master Fund,...",2005180,0002005180-25-000001,200518025000001,D/A,2025-01-31,0002005180-25-000001.txt,2025-Q1,edgar/data/2005180/0002005180-25-000001.txt
4,"0xSecure Fund I, a series of WillowTech Ventur...",1996402,0001996402-25-000001,199640225000001,D/A,2025-01-13,0001996402-25-000001.txt,2025-Q1,edgar/data/1996402/0001996402-25-000001.txt


## Unique Registrant Names

In [17]:
# Count unique registrant names
unique_registrants = df['registrant_name'].nunique()

print(f"Total filings: {len(df):,}")
print(f"Unique registrant names: {unique_registrants:,}")
print(f"Average filings per registrant: {len(df) / unique_registrants:.2f}")

Total filings: 1,231,194
Unique registrant names: 152,402
Average filings per registrant: 8.08


## Top Registrants by Filing Count

In [18]:
# Top 20 most active registrants
top_registrants = df['registrant_name'].value_counts().head(20)

print("Top 20 Most Active Registrants (by number of filings):")
print("=" * 80)
for i, (name, count) in enumerate(top_registrants.items(), 1):
    print(f"{i:2d}. {name[:60]:<60} {count:>6,} filings")

top_registrants

Top 20 Most Active Registrants (by number of filings):
 1. JPMORGAN CHASE & CO                                          21,750 filings
 2. JPMorgan Chase Financial Co. LLC                             20,354 filings
 3. MORGAN STANLEY                                               17,672 filings
 4. Morgan Stanley Finance LLC                                   17,034 filings
 5. UBS AG                                                       11,640 filings
 6. CITIGROUP INC                                                10,713 filings
 7. BANK OF AMERICA CORP /DE/                                    10,200 filings
 8. GOLDMAN SACHS GROUP INC                                      10,114 filings
 9. Citigroup Global Markets Holdings Inc.                       10,098 filings
10. BofA Finance LLC                                              9,663 filings
11. GS Finance Corp.                                              9,104 filings
12. BARCLAYS BANK PLC                                            

registrant_name
JPMORGAN CHASE & CO                       21750
JPMorgan Chase Financial Co. LLC          20354
MORGAN STANLEY                            17672
Morgan Stanley Finance LLC                17034
UBS AG                                    11640
CITIGROUP INC                             10713
BANK OF AMERICA CORP /DE/                 10200
GOLDMAN SACHS GROUP INC                   10114
Citigroup Global Markets Holdings Inc.    10098
BofA Finance LLC                           9663
GS Finance Corp.                           9104
BARCLAYS BANK PLC                          8313
BANK OF MONTREAL /CAN/                     8245
ROYAL BANK OF CANADA                       4404
HSBC USA INC /MD/                          3957
BANK OF NOVA SCOTIA                        3597
TORONTO DOMINION BANK                      3193
iSHARES TRUST                              2767
BlackRock, Inc.                            2640
Innovator ETFs Trust                       1736
Name: count, dtype: int6

## Filing Type Distribution

In [19]:
# Count by form type
form_counts = df['form_type'].value_counts()

print("Form Type Counts (all):")
print("=" * 60)
for form, count in form_counts.items():
    pct = (count / len(df)) * 100
    print(f"{form:<15} {count:>8,} ({pct:>5.2f}%)")

form_counts

Form Type Counts (all):
4                350,987 (28.51%)
424B2            158,844 (12.90%)
144               68,805 ( 5.59%)
8-K               67,650 ( 5.49%)
NPORT-P           52,264 ( 4.24%)
D                 35,143 ( 2.85%)
13F-HR            33,735 ( 2.74%)
SCHEDULE 13G/A    33,419 ( 2.71%)
6-K               27,502 ( 2.23%)
3                 25,923 ( 2.11%)
FWP               24,901 ( 2.02%)
D/A               22,177 ( 1.80%)
497K              21,562 ( 1.75%)
10-Q              17,289 ( 1.40%)
40-APP/A          16,987 ( 1.38%)
SCHEDULE 13G      15,640 ( 1.27%)
497               14,198 ( 1.15%)
N-PX              11,160 ( 0.91%)
10-D              10,872 ( 0.88%)
424B3             10,296 ( 0.84%)
SCHEDULE 13D/A     9,042 ( 0.73%)
ABS-EE             8,876 ( 0.72%)
40-APP             8,507 ( 0.69%)
13F-NT             7,598 ( 0.62%)
DEFA14A            7,269 ( 0.59%)
10-K               6,690 ( 0.54%)
425                6,327 ( 0.51%)
424B5              6,109 ( 0.50%)
CORRESP            6,038

form_type
4            350987
424B2        158844
144           68805
8-K           67650
NPORT-P       52264
              ...  
40FR12G           1
NT 20-F/A         1
T-6               1
6B NTC            1
POS 8C            1
Name: count, Length: 390, dtype: int64

## CIK Analysis

In [20]:
# Unique CIKs
unique_ciks = df['cik'].nunique()

print(f"Unique CIKs: {unique_ciks:,}")
print(f"Unique Registrant Names: {unique_registrants:,}")
print(f"\nDifference: {abs(unique_ciks - unique_registrants):,}")
print("(Same company may have multiple names due to mergers, name changes, etc.)")

Unique CIKs: 151,698
Unique Registrant Names: 152,402

Difference: 704
(Same company may have multiple names due to mergers, name changes, etc.)


## Companies with Multiple Names (Name Changes)

Since there are 151,698 unique CIKs but 152,402 unique registrant names, some companies have filed under multiple names (due to mergers, acquisitions, rebranding, etc.). Let's identify these companies.

In [21]:
# Group by CIK and find CIKs with multiple names
cik_names = df.groupby('cik')['registrant_name'].apply(lambda x: x.unique()).reset_index()
cik_names['name_count'] = cik_names['registrant_name'].apply(len)

# Filter to CIKs with multiple names
multiple_names = cik_names[cik_names['name_count'] > 1].sort_values('name_count', ascending=False)

print(f"CIKs with multiple names: {len(multiple_names):,}")
print(f"CIKs with single name: {len(cik_names[cik_names['name_count'] == 1]):,}")
print(f"\nTop 20 companies with most name variations:")
print("=" * 80)

for idx, row in multiple_names.head(20).iterrows():
    print(f"\nCIK {row['cik']} ({row['name_count']} names):")
    for name in row['registrant_name']:
        print(f"  • {name}")

CIKs with multiple names: 892
CIKs with single name: 150,806

Top 20 companies with most name variations:

CIK 1977837 (4 names):
  • ALIXO-YOLLO Corp
  • ALIXO-YOLLOO CORP.
  • ALIXO-YOLLOO Corp
  • MADE IN USA INC.

CIK 1830974 (3 names):
  • ASPIRIANT RISK-MANAGED CAPITAL APPRECIATION FUND
  • Aspiriant Capital Appreciation Fund
  • Aspiriant Risk-Managed Capital Appreciation Fund

CIK 2051985 (3 names):
  • Iron Horse Acquisition Corp. II
  • Iron Horse Acquisitions Corp. II
  • Iron Horse Acquisition II Corp.

CIK 1897087 (3 names):
  • Junee Ltd
  • Super X AI Technology Ltd
  • SuperX AI Technology Ltd

CIK 1273685 (3 names):
  • NEW YORK MORTGAGE TRUST INC
  • NEW YORK MORTGAGE TRUST, INC.
  • ADAMAS TRUST, INC.

CIK 1829794 (3 names):
  • Volcon, Inc.
  • Empery Digital Inc.
  • Empery Digital, Inc.

CIK 2071668 (3 names):
  • Magnum Ice Cream Company B.V.
  • Magnum Ice Cream Co B.V.
  • Magnum Ice Cream Co N.V.

CIK 1830921 (3 names):
  • ASPIRIANT RISK-MANAGED REAL ASSETS F

In [22]:
# Analyze name changes over time and identify primary (current) name
# Primary name = name used in most recent filing

def show_name_timeline(cik_value):
    """Show how a company's name changed over time"""
    company_filings = df[df['cik'] == cik_value].sort_values('filing_date')
    
    # Get unique names with first and last filing dates
    name_timeline = company_filings.groupby('registrant_name').agg({
        'filing_date': ['min', 'max', 'count']
    }).reset_index()
    name_timeline.columns = ['registrant_name', 'first_filing', 'last_filing', 'filing_count']
    name_timeline = name_timeline.sort_values('first_filing')
    
    # Identify primary name (most recent filing)
    most_recent_filing = company_filings.iloc[-1]
    primary_name = most_recent_filing['registrant_name']
    
    print(f"\nCIK {cik_value} - Name Timeline:")
    print(f"PRIMARY (Current): {primary_name}")
    print("=" * 100)
    for _, row in name_timeline.iterrows():
        is_primary = "[PRIMARY]" if row['registrant_name'] == primary_name else "[ALIAS]  "
        print(f"{is_primary} {row['registrant_name']:<55} {row['first_filing']} to {row['last_filing']} ({row['filing_count']} filings)")
    
    return name_timeline, primary_name

# Show examples for top companies with multiple names
for cik in multiple_names.head(5)['cik'].values:
    show_name_timeline(cik)


CIK 1977837 - Name Timeline:
ALIXO-YOLLOO CORP.                                           2025-01-10 to 2025-01-10 (2 filings)
ALIXO-YOLLO Corp                                             2025-02-07 to 2025-02-07 (1 filings)
ALIXO-YOLLOO Corp                                            2025-03-28 to 2025-09-08 (10 filings)
MADE IN USA INC.                                             2025-09-10 to 2025-12-23 (5 filings)

CIK 1830974 - Name Timeline:
ASPIRIANT RISK-MANAGED CAPITAL APPRECIATION FUND             2025-01-24 to 2025-06-26 (9 filings)
Aspiriant Risk-Managed Capital Appreciation Fund             2025-07-28 to 2025-07-29 (2 filings)
Aspiriant Capital Appreciation Fund                          2025-08-27 to 2025-12-05 (6 filings)

CIK 2051985 - Name Timeline:
Iron Horse Acquisition Corp. II                              2025-01-17 to 2025-01-17 (1 filings)
Iron Horse Acquisitions Corp. II                             2025-05-27 to 2025-08-07 (4 filings)
Iron Horse Acquisition II C

In [23]:
# Distribution of name count
name_count_dist = cik_names['name_count'].value_counts().sort_index()

print("Distribution of Name Counts per CIK:")
print("=" * 60)
for count, freq in name_count_dist.head(10).items():
    pct = (freq / len(cik_names)) * 100
    print(f"{count} name(s): {freq:>8,} CIKs ({pct:>5.2f}%)")

if len(name_count_dist) > 10:
    print(f"...")
    print(f"Max names for single CIK: {name_count_dist.index.max()}")

Distribution of Name Counts per CIK:
1 name(s):  150,806 CIKs (99.41%)
2 name(s):      878 CIKs ( 0.58%)
3 name(s):       13 CIKs ( 0.01%)
4 name(s):        1 CIKs ( 0.00%)


## Primary Name Determination Logic

In [ ]:
# Explain the logic for determining primary name
print("How to Determine Primary (Current) Name for a CIK:")
print("=" * 80)
print("\n1. PRIMARY NAME = Name used in the MOST RECENT filing")
print("   - This represents the company's current legal name")
print("   - Most reliable indicator of current identity")
print("\n2. ALIAS NAMES = All other historical names")
print("   - Previous legal names (before rebranding/merger)")
print("   - Should be stored in company.aliases[] array")
print("\n3. Example Timeline:")
print("   2020-2021: 'Facebook, Inc.' [ALIAS]")
print("   2021-2025: 'Meta Platforms, Inc.' [PRIMARY]")
print("   → Primary: 'Meta Platforms, Inc.'")
print("   → Aliases: ['Facebook, Inc.']")
print("\n4. Edge Cases:")
print("   - If names overlap in time: Use most recent filing date")
print("   - If multiple names on same date: Use name with most filings on that date")
print("   - If still tied: Alphabetically first (deterministic)")

# Show statistics on primary vs alias names
print("\n" + "=" * 80)
print("Statistics:")
print(f"  Total CIKs with multiple names: {len(multiple_names):,}")
print(f"  Total name records: {multiple_names['name_count'].sum():,}")
print(f"  Primary names (1 per CIK): {len(multiple_names):,}")
print(f"  Alias names: {multiple_names['name_count'].sum() - len(multiple_names):,}")
print(f"  Average aliases per CIK: {(multiple_names['name_count'].sum() - len(multiple_names)) / len(multiple_names):.2f}")

In [24]:
# Export CIKs with multiple names to CSV for further analysis
output_path = Path('../output/registrant_metadata_2025_multiple_names.csv')

# Create detailed export with all names per CIK, marking primary name
export_data = []
for _, row in multiple_names.iterrows():
    cik = row['cik']
    names = row['registrant_name']
    
    # Get filing dates for each name
    company_filings = df[df['cik'] == cik]
    
    # Identify primary name (most recent filing)
    most_recent_filing = company_filings.sort_values('filing_date').iloc[-1]
    primary_name = most_recent_filing['registrant_name']
    
    for name in names:
        name_filings = company_filings[company_filings['registrant_name'] == name]
        export_data.append({
            'cik': cik,
            'registrant_name': name,
            'is_primary': name == primary_name,
            'first_filing_date': name_filings['filing_date'].min(),
            'last_filing_date': name_filings['filing_date'].max(),
            'filing_count': len(name_filings),
            'total_names_for_cik': row['name_count']
        })

export_df = pd.DataFrame(export_data)
export_df = export_df.sort_values(['cik', 'is_primary', 'first_filing_date'], ascending=[True, False, True])
export_df.to_csv(output_path, index=False)

print(f"\nExported {len(export_df):,} name records for {len(multiple_names):,} CIKs to:")
print(f"{output_path}")
print(f"\nPrimary names: {export_df['is_primary'].sum():,}")
print(f"Alias names: {(~export_df['is_primary']).sum():,}")
print(f"\nSample of export (showing primary names first):")
export_df.head(20)


Exported 1,799 name records for 892 CIKs to:
../output/registrant_metadata_2025_multiple_names.csv

Sample of export:


,cik,registrant_name,first_filing_date,last_filing_date,filing_count,total_names_for_cik
141,9211,ROBERT W. BAIRD & CO. INCORPORATED,2025-02-04,2025-08-14,31,2
142,9211,ROBERT W. BAIRD & CO. Inc,2025-09-18,2025-12-11,11,2
1023,19481,"Virtus Investment Advisers, Inc.",2025-02-14,2025-02-14,1,2
1024,19481,"Virtus Investment Advisers, LLC",2025-03-07,2025-12-12,11,2
1195,26780,DANA INC,2025-01-23,2025-03-14,47,2
1196,26780,DANA Inc,2025-03-11,2025-12-04,87,2
1197,28412,COMERICA INC /NEW/,2025-01-22,2025-08-25,102,2
1198,28412,COMERICA INC,2025-09-02,2025-12-30,67,2
1199,36840,FIRST REAL ESTATE INVESTMENT TRUST OF NEW JERSEY,2025-01-17,2025-04-03,16,2
1200,36840,FIRST REAL ESTATE INVESTMENT TRUST OF NEW JERS...,2025-05-29,2025-10-08,9,2


## Notable Examples: Well-Known Name Changes

In [ ]:
# Search for well-known companies that changed names
notable_searches = [
    ('Meta', 'Facebook'),
    ('Alphabet', 'Google'),
    ('X Corp', 'Twitter'),
    ('Warner Bros', 'Time Warner'),
]

print("Notable Name Changes:")
print("=" * 100)

for search1, search2 in notable_searches:
    # Find CIKs that have both names
    matches1 = df[df['registrant_name'].str.contains(search1, case=False, na=False)]['cik'].unique()
    matches2 = df[df['registrant_name'].str.contains(search2, case=False, na=False)]['cik'].unique()
    
    common_ciks = set(matches1) & set(matches2)
    
    if common_ciks:
        print(f"\nFound CIK(s) with both '{search1}' and '{search2}':")
        for cik in common_ciks:
            show_name_timeline(cik)
    else:
        print(f"\nNo CIK found with both '{search1}' and '{search2}'")
        if matches1.any():
            print(f"  Found {len(matches1)} CIK(s) with '{search1}'")
        if matches2.any():
            print(f"  Found {len(matches2)} CIK(s) with '{search2}'")

## Same Name, Multiple CIKs

The reverse case: same registrant name used by multiple CIKs. This could indicate:
- Different legal entities with similar/identical names
- Subsidiaries with same name in different jurisdictions
- Data quality issues or name collisions

In [25]:
# Group by registrant name and find names with multiple CIKs
name_ciks = df.groupby('registrant_name')['cik'].apply(lambda x: x.unique()).reset_index()
name_ciks['cik_count'] = name_ciks['cik'].apply(len)

# Filter to names with multiple CIKs
multiple_ciks = name_ciks[name_ciks['cik_count'] > 1].sort_values('cik_count', ascending=False)

print(f"Names with multiple CIKs: {len(multiple_ciks):,}")
print(f"Names with single CIK: {len(name_ciks[name_ciks['cik_count'] == 1]):,}")
print(f"\nTop 20 names with most CIK variations:")
print("=" * 80)

for idx, row in multiple_ciks.head(20).iterrows():
    print(f"\n'{row['registrant_name']}' ({row['cik_count']} CIKs):")
    for cik in row['cik']:
        filing_count = len(df[(df['registrant_name'] == row['registrant_name']) & (df['cik'] == cik)])
        print(f"  • CIK {cik} - {filing_count} filings")

Names with multiple CIKs: 164
Names with single CIK: 152,238

Top 20 names with most CIK variations:

'LaSalle Global Employee Co-Investment Fund B L.P., in respec' (6 CIKs):
  • CIK 2003103 - 1 filings
  • CIK 2003109 - 1 filings
  • CIK 2061075 - 1 filings
  • CIK 2061087 - 1 filings
  • CIK 2061303 - 1 filings
  • CIK 2061311 - 1 filings

'Partners Group Private Credit CV S.C.A., SICAV-RAIF - PG 202' (6 CIKs):
  • CIK 2077563 - 1 filings
  • CIK 2077565 - 1 filings
  • CIK 2077566 - 1 filings
  • CIK 2077567 - 1 filings
  • CIK 2077569 - 1 filings
  • CIK 2077571 - 1 filings

'Silver Point Distressed Opportunity Institutional Partners I' (5 CIKs):
  • CIK 1949011 - 1 filings
  • CIK 1949012 - 1 filings
  • CIK 1968475 - 1 filings
  • CIK 1968476 - 1 filings
  • CIK 1968477 - 1 filings

'OHA CA Customized Credit Fund, L.P. - OHA Senior Private Len' (4 CIKs):
  • CIK 1892221 - 3 filings
  • CIK 1892222 - 3 filings
  • CIK 2025394 - 1 filings
  • CIK 2025395 - 1 filings

'Brookfield As

In [26]:
# Distribution of CIK count per name
cik_count_dist = name_ciks['cik_count'].value_counts().sort_index()

print("Distribution of CIK Counts per Name:")
print("=" * 60)
for count, freq in cik_count_dist.head(10).items():
    pct = (freq / len(name_ciks)) * 100
    print(f"{count} CIK(s): {freq:>8,} names ({pct:>5.2f}%)")

if len(cik_count_dist) > 10:
    print(f"...")
    print(f"Max CIKs for single name: {cik_count_dist.index.max()}")

Distribution of CIK Counts per Name:
1 CIK(s):  152,238 names (99.89%)
2 CIK(s):      140 names ( 0.09%)
3 CIK(s):       14 names ( 0.01%)
4 CIK(s):        7 names ( 0.00%)
5 CIK(s):        1 names ( 0.00%)
6 CIK(s):        2 names ( 0.00%)


In [34]:
# Analyze timeline for names with multiple CIKs
def show_cik_timeline(name):
    """Show how different CIKs used the same name over time"""
    name_filings = df[df['registrant_name'] == name].sort_values('filing_date')
    
    # Get unique CIKs with first and last filing dates
    cik_timeline = name_filings.groupby('cik').agg({
        'filing_date': ['min', 'max', 'count']
    }).reset_index()
    cik_timeline.columns = ['cik', 'first_filing', 'last_filing', 'filing_count']
    cik_timeline = cik_timeline.sort_values('first_filing')
    
    print(f"\n'{name}' - CIK Timeline:")
    print("=" * 100)
    for _, row in cik_timeline.iterrows():
        print(f"CIK {row['cik']:<15} {row['first_filing']} to {row['last_filing']} ({row['filing_count']} filings)")
    
    return cik_timeline

# Show examples for top names with multiple CIKs
for name in multiple_ciks.head(5)['registrant_name'].values:
    show_cik_timeline(name)


'LaSalle Global Employee Co-Investment Fund B L.P., in respec' - CIK Timeline:
CIK 2003103         2025-03-13 to 2025-03-13 (1 filings)
CIK 2003109         2025-03-13 to 2025-03-13 (1 filings)
CIK 2061075         2025-04-28 to 2025-04-28 (1 filings)
CIK 2061087         2025-04-28 to 2025-04-28 (1 filings)
CIK 2061303         2025-04-28 to 2025-04-28 (1 filings)
CIK 2061311         2025-04-28 to 2025-04-28 (1 filings)

'Partners Group Private Credit CV S.C.A., SICAV-RAIF - PG 202' - CIK Timeline:
CIK 2077563         2025-07-21 to 2025-07-21 (1 filings)
CIK 2077565         2025-07-21 to 2025-07-21 (1 filings)
CIK 2077566         2025-07-21 to 2025-07-21 (1 filings)
CIK 2077567         2025-07-21 to 2025-07-21 (1 filings)
CIK 2077569         2025-07-21 to 2025-07-21 (1 filings)
CIK 2077571         2025-07-21 to 2025-07-21 (1 filings)

'Silver Point Distressed Opportunity Institutional Partners I' - CIK Timeline:
CIK 1949011         2025-07-31 to 2025-07-31 (1 filings)
CIK 1949012        

In [28]:
# Export names with multiple CIKs to CSV
output_path = Path('../output/registrant_metadata_2025_multiple_ciks.csv')

# Create detailed export with all CIKs per name
export_data = []
for _, row in multiple_ciks.iterrows():
    name = row['registrant_name']
    ciks = row['cik']
    
    # Get filing dates for each CIK
    name_filings = df[df['registrant_name'] == name]
    
    for cik in ciks:
        cik_filings = name_filings[name_filings['cik'] == cik]
        export_data.append({
            'registrant_name': name,
            'cik': cik,
            'first_filing_date': cik_filings['filing_date'].min(),
            'last_filing_date': cik_filings['filing_date'].max(),
            'filing_count': len(cik_filings),
            'total_ciks_for_name': row['cik_count']
        })

export_df = pd.DataFrame(export_data)
export_df = export_df.sort_values(['registrant_name', 'first_filing_date'])
export_df.to_csv(output_path, index=False)

print(f"\nExported {len(export_df):,} CIK records for {len(multiple_ciks):,} names to:")
print(f"{output_path}")
print(f"\nSample of export:")
export_df.head(20)


Exported 367 CIK records for 164 names to:
../output/registrant_metadata_2025_multiple_ciks.csv

Sample of export:


,registrant_name,cik,first_filing_date,last_filing_date,filing_count,total_ciks_for_name
102,AB European Commercial Real Estate Debt (Luxem...,2011108,2025-02-21,2025-02-21,1,2
101,AB European Commercial Real Estate Debt (Luxem...,1966221,2025-02-28,2025-02-28,1,2
39,AB Private Credit Investors Middle Market Dire...,1801524,2025-01-27,2025-10-03,4,4
37,AB Private Credit Investors Middle Market Dire...,1634385,2025-03-18,2025-10-03,5,4
38,AB Private Credit Investors Middle Market Dire...,1746553,2025-03-18,2025-10-08,5,4
40,AB Private Credit Investors Middle Market Dire...,1746552,2025-07-11,2025-07-11,1,4
84,"ARENA SHORT DURATION HIGH YIELD (CAYMAN) FUND,...",1699844,2025-02-06,2025-02-06,1,3
85,"ARENA SHORT DURATION HIGH YIELD (CAYMAN) FUND,...",1699845,2025-02-06,2025-02-06,1,3
86,"ARENA SHORT DURATION HIGH YIELD (CAYMAN) FUND,...",1699846,2025-02-06,2025-02-06,1,3
261,Agrawal Ajay,1808151,2025-02-05,2025-06-02,3,2


## Summary: Name-CIK Relationships

In [36]:
# Summary of both analyses
print("Name-CIK Relationship Summary")
print("=" * 80)
print(f"\nOne-to-One (1 CIK = 1 Name):")
one_to_one = len(cik_names[cik_names['name_count'] == 1])
print(f"  {one_to_one:,} CIKs have exactly one name")

print(f"\nOne-to-Many (1 CIK = Multiple Names):")
print(f"  {len(multiple_names):,} CIKs have multiple names (name changes/rebranding)")
print(f"  Total name variations: {multiple_names['name_count'].sum():,}")
print(f"  Average names per CIK: {multiple_names['name_count'].mean():.2f}")
print(f"  Max names for single CIK: {multiple_names['name_count'].max()}")

print(f"\nMany-to-One (Multiple CIKs = 1 Name):")
print(f"  {len(multiple_ciks):,} names are used by multiple CIKs (different entities)")
print(f"  Total CIK variations: {multiple_ciks['cik_count'].sum():,}")
print(f"  Average CIKs per name: {multiple_ciks['cik_count'].mean():.2f}")
print(f"  Max CIKs for single name: {multiple_ciks['cik_count'].max()}")

print(f"\nData Quality Metrics:")
print(f"  Unique CIKs: {unique_ciks:,}")
print(f"  Unique Names: {unique_registrants:,}")
print(f"  Difference: {abs(unique_ciks - unique_registrants):,}")
print(f"  Name-to-CIK ratio: {unique_registrants / unique_ciks:.4f}")

Name-CIK Relationship Summary

One-to-One (1 CIK = 1 Name):
  150,806 CIKs have exactly one name

One-to-Many (1 CIK = Multiple Names):
  892 CIKs have multiple names (name changes/rebranding)
  Total name variations: 1,799
  Average names per CIK: 2.02
  Max names for single CIK: 4

Many-to-One (Multiple CIKs = 1 Name):
  164 names are used by multiple CIKs (different entities)
  Total CIK variations: 367
  Average CIKs per name: 2.24
  Max CIKs for single name: 6

Data Quality Metrics:
  Unique CIKs: 151,698
  Unique Names: 152,402
  Difference: 704
  Name-to-CIK ratio: 1.0046


## Quarterly Breakdown

In [30]:
# Filings by quarter
quarterly = df['source_quarter'].value_counts().sort_index()

print("Filings by Quarter:")
print("=" * 40)
for quarter, count in quarterly.items():
    print(f"{quarter}: {count:>10,} filings")

quarterly

Filings by Quarter:
2025-Q1:    339,748 filings
2025-Q2:    331,218 filings
2025-Q3:    285,256 filings
2025-Q4:    274,972 filings


source_quarter
2025-Q1    339748
2025-Q2    331218
2025-Q3    285256
2025-Q4    274972
Name: count, dtype: int64